File: tinysat-python/base/instance.py
class Instance:
    def __init__(self, variables, clauses):
        """
        self.variables: [1, 2, 3, 4]
        self.clauses: [(0, 2, 5), (3, 4, 5)]
        """
        self.var_count = len(variables)
        self.clause_count = len(clauses)
        self.variables = variables
        self.clauses = clauses

    def __str__(self):
        text = f"Variables: {self.var_count}\n{self.variables}\n"
        text += f"Clauses: {self.clause_count}\n{self.clauses}\n"
        return text
    

File: tinysat-java/src/tinysat/base/Instance.java
package tinysat.base;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Iterator;
import java.util.List;

public class Instance {
    private int varCount;

    private List<List<Integer>> clauses;

    public Instance(int varCount, List<List<Integer>> clauses) {
        this.varCount = varCount;
        this.clauses = clauses;
    }

    public int getVarCount() {
        return varCount;
    }

    public List<List<Integer>> getClauses() {
        return clauses;
    }

    public List<Integer> getClause(Integer i) {
        return clauses.get(i);
    }
}
